docs/new_flux_ideas_and_guide.md

```diff
@@ -0,0 +1,985 @@
+h2o_temperature,location=santa_monica,state=CA surface_degrees=61,bottom_degrees=48 2020-02-22T17:01
+h2o_temperature,location=santa_monica,state=CA surface_degrees=60,bottom_degrees=48 2020-02-22T17:31
+h2o_temperature,location=santa_monica,state=CA surface_degrees=60,bottom_degrees=48 2020-02-22T17:46
+h2o_temperature,location=coyote_creek,state=CA surface_degrees=55,bottom_degrees=51 2020-02-22T16:05
+h2o_temperature,location=coyote_creek,state=CA surface_degrees=53,bottom_degrees=50 2020-02-22T17:05
+h2o_temperature,location=puget_sound,state=WA surface_degrees=55,bottom_degrees=40 2020-02-22T17:10
+h2o_temperature,location=puget_sound,state=WA surface_degrees=54,bottom_degrees=40 2020-02-22T17:40
+```
+
+In the data set we can see there are three distinct keys and that the time range of data is from 
+`2020-02-22T15:01` to `2020-02-22T17:46` (about a two and a half hour span). There are more data 
+points for the `santa_monica`, `CA` location than for the others. The differing number of values 
+will be instructive later when we shape and summarize this data.
+
+#### Summarizing and Grouping with Aggregates
+Let's get into an example for summarizing some of our water data. For the sake of these queries, assume that 
+we're executing them at `2020-02-22T18:00`, that way we can use relative times in the query. Say we want to get the 
+last 3 hours of data, with the min, max, and mean of `bottom_degrees` for that time period. With that 
+definition, we'd expect to get 1 row per series in the dataset (assuming that we have min, max, and mean as 
+columns in the resulting table. Here's the query:
+
+```
+@my_data.h2o_temperature{bottom_degrees, time > -3h}
+    |> aggregate({min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)}) 
+```
+
+Will return the following dataset:
+
+| min_bottom_degrees | max_bottom_degrees | mean_bottom_degrees |
+| --- | --- | --- |
+| 48 | 51 | 47.45 |
+
+Even though the sample data set has three series, we get back only a single row that gives the summary statistics we 
+requested. The call to the aggregate function passed in an **Aggregate Summary Block** which looks similar in 
+construction to the **Predicate Filter Block** that we used on the first line of the query. The returned column names 
+are given automatically based on the construction `<function name>_<field name>`. As with the other blocks, field 
+names can be wrapped in double quotes to handle cases like spaces or other special characters in a field name.
+
+Our results don't have the descriptive tag metadata because we didn't ask for it. Selecting `location` and `state` 
+which are the two tags that we have for data in the `h20_measurements` measurement will ensure it comes through in 
+the aggregate and we can then group by those columns:
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees, time > -3h}
+    |> aggregate({min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)}, by: ["location", "state"]) 
```

Comment:
I'd try to write this first as:

`@my_data.h2o_temperature{location, state, min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)} |> group(by: ["location", "state"]`

Reply:
Hah, you could try but it wouldn't be right. Ideally the parser would return a helpful error that leads you to calling `min` with `select` or `aggregate`. You should view each function as taking an input and doing something with it.